tools/dataset-aggregator/aggregator.py

```diff
@@ -0,0 +1,65 @@
+import json
+import os
+import sys
+from shutil import move
+from origo.devportal.poctools.models import Dataset
+from origo.devportal.poctools.models import Visibility
+
+AGGREGATIONS_PATH = os.environ['AGGREGATIONS_PATH']
+RESULT_PATH = os.environ['RESULT_PATH']
+TMP_PATH = os.environ['TMP_PATH']
+
+VISIBILITY_PATH = os.path.join(AGGREGATIONS_PATH, 'visibility.json')
+
+def aggregate(datasets, visibility_map=dict):
+    aggregated_datasets = list()
+
+    for dataset in datasets:
+        aggregated = Dataset()
+        aggregated.identifier = dataset["identifier"]
+        aggregated.title = dataset["title"]
+        aggregated.description = dataset.get("description", "")
+        aggregated.publisher = dataset.get("publisher", "")
+        aggregated.visibility = visibility_map.get(aggregated.identifier, Visibility.PRIVATE.value)
+
+        aggregated_datasets.append(aggregated)
+
+    return aggregated_datasets
+
+
+def divideByVisibility(data):
+    result = { level: list() for level in Visibility }
+
+    for item in data:
+        # Output shouldn't contain visibility. It's internal information for our domain.
```

Comment:
I think I decided to filter out this information in latest_provider due to us might wanting the information in some kind of authenticated frontend in the future. Future proofing, I know.

Reply:
I'm debating back and forth if the view layer (the latest_provider) or the business logic (the aggregator) should deal with this. For now I agree it makes for the view layer to do this filtering, so I'll this there as well.